components-registry-service-server/src/main/kotlin/org/octopusden/octopus/components/registry/server/service/impl/EmployeeDirectoryService.kt

```
@@ -148,6 +148,23 @@ class EmployeeDirectoryService(
     fun probe(): IntegrationHealth = probeHealth().health
 
+    /**
+     * Returns the manager's sAMAccountName for [ownerUsername], or null when the
+     * user has no manager, is not found, or the service is unavailable (fail-open).
+     */
+    @Suppress("TooGenericExceptionCaught")
+    fun getManager(ownerUsername: String): String? {
+        val client = clientProvider.getIfAvailable() ?: return null
+        return try {
+            client.getManager(ownerUsername).manager
+        } catch (e: NotFoundException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)